scripts/install-darwin-multi-user.sh

```diff
@@ -6,7 +6,7 @@ set -o pipefail
 readonly NIX_DAEMON_DEST=/Library/LaunchDaemons/org.nixos.nix-daemon.plist
 # create by default; set 0 to DIY, use a symlink, etc.
 readonly NIX_VOLUME_CREATE=${NIX_VOLUME_CREATE:-1} # now default
-NIX_FIRST_BUILD_UID="301"
+NIX_FIRST_BUILD_UID="${NIX_FIRST_BUILD_UID:-301}"
```

Comment:
We can use a different var name to avoid it being overridden by the parent script.

```suggestion
NIX_FIRST_BUILD_UID="${NIX_FIRST_BUILD_UID_DARWIN:-301}"
```

Reply:
While it is the simpler change, I think the potential-confusion would be lower if we moved the value-setting out into the ~speciated scripts (i.e., require the platform install scripts to be opinionated on this UID instead of having a default-that-isn't-a-default?)

Reply:
I'm not sure what you mean.

Reply:
Something like:
- replace the default declaration up in the main multiuser script with a comment that says that the speciated ~variant (`-darwin-`, `-linux-`) scripts are responsible for setting it 
- move the default declaration out into the `-linux-` script
- do roughly what this current PR does for the `-darwin-` script

Reply:
gotcha. yeah, sounds good.